File: 程序员面试金典/1.5一次编辑/oneEditAway.java
class Solution {
    public boolean oneEditAway(String first, String second) {
        if (first.length() < second.length()) {
            String temp = first;
            first = second;
            second = temp;
        }
        if (first.length() - second.length() >= 2)
            return false;
        int i = 0, j = 0;
        boolean foundDiff = false;
        while (i < first.length() && j < second.length()) {
            if (first.charAt(i) != second.charAt(j)) {
                if (foundDiff)
                    return false;
                foundDiff = true;
                if (first.length() == second.length())
                    ++j;
            }
            else
                ++j;
            ++i;
        }
        return true;
    }
}

File: 程序员面试金典/1.5/oneEditAway.py
class Solution:
    def oneEditAway(self, first: str, second: str) -> bool:
        if len(first) < len(second):
            first, second = second, first
        if len(first) - len(second) >= 2:
            return False
        i = 0
        while i < len(second) and first[i] == second[i]:
            i += 1
        return first[i + 1:] == second[i:] or first[i + 1:] == second[i + 1:]
